lib/Plack/Middleware/MethodOverride.pm

```diff
@@ -3,17 +3,17 @@ package Plack::Middleware::MethodOverride;
 use strict;
 use 5.8.1;
 use parent qw(Plack::Middleware);
+use Plack::Util::Accessor 'param';
 use URI;
 our $VERSION = '0.12';
 
 my %ALLOWED = map { $_ => undef } qw(GET HEAD PUT DELETE OPTIONS TRACE CONNECT);
 
 sub new {
-    my $self = shift->SUPER::new(
-        param  => 'x-tunneled-method',
-        (@_ == 1 && ref $_[0] eq 'HASH' ? %{ +shift } : @_),
-    );
-    $self->header($self->header || 'X-HTTP-Method-Override');
+    my $class = shift;
+    my $self = $class->SUPER::new(@_);
+    $self->param('x-tunneled-method') if not exists $self->{param};
+    $self->header(exists $self->{header} ? $self->{header} : 'X-HTTP-Method-Override');
```

Comment:
Hrm. Maybe should be

    $self->header('X-HTTP-Method-Override') if not exists $self->{header}

?